### ISSUE TITLE
Ch 23 - Error when posting new user via API due to no last_seen entry

### ISSUE THREAD

**Issue by rlheureu** [miguelgrinberg/microblog#81]
I'm up to the part where we're testing our new way of creating users, with the instruction to create Alice:

`
venv) $ http POST http://localhost:5000/api/users username=alice password=dog \
    email=alice@example.com "about_me=Hello, my name is Alice!"
`

I am seeing this error, which I interpret as being due to the app expecting a "last seen" time with any new user.

```[2018-02-05 14:43:01,243] ERROR in app: Exception on /api/users [POST]
Traceback (most recent call last):
  File "~/venv/lib/python2.7/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "~/venv/lib/python2.7/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "~/venv/lib/python2.7/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "~/venv/lib/python2.7/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "~/venv/lib/python2.7/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "~/app/api/users.py", line 45, in create_user
    response = jsonify(user.to_dict())
  File "~/app/models.py", line 171, in to_dict
    'last_seen': self.last_seen.isoformat() + 'Z',
AttributeError: 'NoneType' object has no attribute 'isoformat'
```

Running with my theory, I confirmed it is creating the user in the database, so I attempted logging in as Alice, hoping it would assign a "last seen"; however, the app simply continued throwing errors.

**Comment by miguelgrinberg** [comment]
The `last_seen` field has a default assigned to it (here: https://github.com/miguelgrinberg/microblog/blob/master/app/models.py#L91), so whenever you create a new user the current time should be assigned to this field. Maybe you missed that in your copy of the application?

**Comment by rlheureu** [comment]
Ah, you're right, and it seems to be a versioning problem.

That line was added to the code in Chapter 6: https://learn.miguelgrinberg.com/read/mega-tutorial/ch06.html#cid50

In Chapter 15, I copied your code over, so I was sure I had the right Blueprint. In your version of Chapter 15, it's missing the assignment: https://github.com/miguelgrinberg/microblog/blob/v0.15/app/models.py#L25.

In fact, I went through many of the versions of models.py and it doesn't seem to appear again until your Ch23 version.
